tsconfig.json

```diff
@@ -1,24 +1,29 @@
 {
 	"compilerOptions": {
 		"module": "commonjs",
-		"target": "es6",
+		"target": "ES2022",
 		"outDir": "out",
 		"lib": [
-			"es6",
+			"ES2022",
 			"dom"
 		],
 		"sourceMap": true,
-		"rootDir": "src",
 		"strict": true,
+		"esModuleInterop": true,
 		"noImplicitReturns": true,
 		"allowSyntheticDefaultImports": true,
 		"resolveJsonModule": true,
+		"experimentalDecorators": true,
```

Comment:
hmm, so this whole thing is experimental?

Reply:
Decorators are a stage 3 proposal in ECMAScript (https://github.com/tc39/proposal-decorators), so they will be part of the ECMAScript standard soon. Transpilers like Typescript and Babel allow you to already use them based on the specifications decided on in the stages 1-3 of the feature proposal. So they are only 'experimental' in some sense, it's more like a preview and the 'experimental' name indicates that there may be slight changes in how they behave. We, here, are using the core mechanics though - so what's unlikely to change.